config/config.go

```diff
@@ -238,6 +244,30 @@ func (c Config) FindComponent(name string) *resource.Config {
 	return nil
 }
 
+// setUnprocessedConfig updates unprocessedConfig.
+func (c *Config) setUnprocessedConfig(cfg *Config) {
+	c.unprocessedConfig = cfg
+}
+
+// UnprocessedConfig returns unprocessedConfig.
+func (c Config) UnprocessedConfig() *Config {
```

Comment:
nit: for consistencies sake, we should probably make this a pointer-receiver and internally make a copy of the config.